## 6. Feladatsor

<img src="phd022509s.jpg" width="700">

<br>
### Egyszerű iteráció (fixpont iteráció)
1.feladat. Mutassuk meg, hogy a $\varphi (x)=\sqrt{x},\ x\in(1,2)$ leképezés kontrakció, de nincs fixpontja.<br>

2.feladat. Tekintsük az alábbi függvényt:
$$
\displaystyle \varphi(x)=x+\frac{1}{1+x},\ x\in[0,\infty).
$$
Igazoljuk, hogy $|\varphi (x)-\varphi(y)|<|x-y|$ teljesül minden $x,y\in[0,\infty)$ esetén, viszont $\varphi$-nek mégsincs fixpontja.<br>

3.feladat. Alkalmazzuk az egyszerű iterációt és tegyünk meg vele négy lépést. Ellenőrizzük, hogy teljesülnek-e a fixponttétel feltételei. Amennyiben igen, akkor az $x_0=0$ pontból hány iterációs lépést kell megtennünk ahhoz, hogy $10^{-5}$ nagyságú hibával határozzuk meg az egyenlet megoldását, ha<br>

(a) $x=\sqrt{x+2}$, $x\in[0,2]$<br>
(b) $x=\cos(x)$, $x\in[0,1]$<br>


In [17]:
% 3.feladat (a) resze egyszeru iteracio segitsegevel
clear all;
x=0; fun='cos(x)'; kmax=1000;
format long
k=0;
while k<kmax
   k=k+1;
   x=eval(fun);
end;
x

% Hany lepes szukseges?
tol=1e-5;
x_0=0;
q=sin(1);
x_1=1;
n=log(((1-q)*tol)/(abs(x_1-x_0)))/log(q)
ceil(n);
disp(ceil(n))

x =

   0.739085133215161


n =

  77.372266795527878

    78


<br>

4.feladat. Legyen $\varphi\in C^s(I),\ \varphi^{(j)}(x^*)=0$ minden $j=1,\ldots,s-1$-re, de $\varphi^{(s)}(x^*)\neq 0$. Ekkor az egyszerû iterációval kapott $(x_n)$ sorozatra igaz, hogy $x_n\to x^*$ $s$-edrendben.<br>

5.feladat. (Beadható) Az $x-\cos (x)=0,\ x\in[0,1]$ egyenlet megoldásának közelítéséhez tegyünk meg néhány lépést Aitken módszerével és az Aitken-Steffensen iterációval is az $x_0=0$ kezdõértékkel. Hasonlítsuk össze a kapott eredményeket a pontos $x^*=0.739085133215$ értékkel.<br>

6.feladat. Az alábbi $\varphi(x)=x$ alakok melyikére érdemes hozni az $x^2-2=0,\ x\in[1,2]$ egyenletet, hogy az egyszerû iterációt alkalmazni tudjuk?<br>

(a) $\displaystyle x=\frac{2}{x}$<br>
(b) $\displaystyle x=1+\frac{1}{1+x}$ <br>
(c) $\displaystyle x=\frac{1}{2}\Big(x+\frac{2}{x}\Big)$<br>

Tegyünk meg négy lépést az átírásokból származtatható $x_{n+1}=\varphi(x_n)$ iterációk mindegyikével. Magyarázzuk meg, hogy mi okozza a módszerek közötti konvergencia rend eltérését.<br>

7.feladat. Hozzuk az $x^3-x=1000,\ x\in[10,11]$ egyenletet olyan alakra, hogy egyszerû iterációval megoldható legyen.<br>

### Klasszikus iterációk (intervallumfelezés, húr- és szelőmódszer, Newton-módszer

8.feladat. Oldjuk meg az $x^2-x-6=0,\ x\in[1,4]$ egyenletet. Tegyünk meg három lépést a klasszikus módszerekkel (egyszerû iteráció, intervallumfelezés, húrmódszer, szelõmódszer) és a Newton-módszerrel.<br>


In [40]:
% Intervallumfelezes
clear all
a=1; b=4; func='x^2-x-6'; kmax=4; toll=10^-10;

format long
k=0;
while k<kmax & b-a>toll
   k=k+1;
   x=a+(b-a)/2;
      fx=eval(func);
   if fx>0 
      b=x;
   else
      a=x;
   end;
end;
x



x =

   3.062500000000000


In [54]:
% Húrmódszer
clear all;
format long

a=1; b=4; func='x^2-x-6'; nmax=3; toll=10^-10; 

x=a; f=eval(func); fx=[f]; x=b; f=eval(func); fx=[fx, f]; 
nit=0;     
xvect=[a,b]; xdif=[]; f=toll+1; kprime=1;
while (nit < nmax) & (abs(f) > toll),
nit=nit+1; dim=length(xvect);
x=xvect(dim); fxk=eval(func); xk=x; i=dim;
while (i >= kprime), i=i-1; x=xvect(i); fxkpr=eval(func);
if ((fxkpr*fxk) < 0), xkpr=x; kprime=i; break; end;
end;
x=xk-fxk*(xk-xkpr)/(fxk-fxkpr); xvect=[xvect, x]; f=eval(func);
fx=[fx, f]; err=abs(x-xkpr); xdif=[xdif, err];
end;
xvect;
xvect(2+nmax)

ans =

   2.984615384615385


In [11]:
clear all;
func='x^2-x-6'; kmax=4; toll=10^-10;
x=1; fxold=eval(func); xold=x;
x=4; fx=eval(func);
format long
err=toll+1;
k=0;
while k<kmax & err>toll
   k=k+1;
   xnew=x-(xold-x)*fx/(fxold-fx);
   xold=x;
   fxold=eval(func);
   x=xnew
   fx=eval(func);
   err=abs(x-xold);
end;
x



x =

   2.500000000000000


x =

   2.909090909090909


x =

   3.010309278350515


x =

   2.999809487521433


x =

   2.999809487521433


In [12]:
% Newton-modszer

clear all;
%x=2; func='x^3+2*x^2+10*x-20'; funcder='3*x^2+4*x+10'; kmax=20; toll=10^-10;
x=sqrt(5)/5; func='x^3-x'; funcder='3*x^2-1'; kmax=100; toll=10^-10;
format long
err=toll+1;
k=0;
while k<kmax & err>toll
   k=k+1;
   xold=x;
   x=x-eval(func)/eval(funcder);
   err=abs(x-xold);
end;
x
format short


x =

    -1


<br>

9.feladat. Az $x^3-x=0$ egyenlet $x^*=0$ gyökét szeretnénk Newton-módszerrel közelíteni. Az $x^*$ mekkora környezetébõl lehet az iterációt indítani, hogy biztosan konvergáljon a módszer? Milyen helyzet áll elõ, ha $\displaystyle x_0=\frac{\sqrt{5}}{5}$-tel indulunk?